api/src/test/java/io/kafbat/ui/OpenLdapPIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+
+  @Autowired
+  private WebTestClient webTestClient;
+
+  @DynamicPropertySource
+  static void neo4jProperties(DynamicPropertyRegistry registry) {
+    registry.add("spring.ldap.urls", LDAP_CONTAINER::getLdapUrl);
+  }
+
+  @BeforeAll
+  static void setup() {
+    LDAP_CONTAINER.start();
+  }
+
+  @AfterAll
+  static void shutdown() {
+    LDAP_CONTAINER.stop();
+  }
+
+  @Test
+  public void testUserPermissions() {
+    AuthenticationInfoDTO info = authenticationInfo("johndoe");
+
+    assertNotNull(info);
+    assertTrue(info.getRbacEnabled());
+    System.out.println("info = " + info);
+    List<UserPermissionDTO> permissions = info.getUserInfo().getPermissions();
+    assertFalse(permissions.isEmpty());
+    assertTrue(permissions.stream().anyMatch(permission ->
+        permission.getClusters().contains(LOCAL) && permission.getResource() == ResourceTypeDTO.TOPIC));
+    assertEquals(permissions, authenticationInfo("johnwick").getUserInfo().getPermissions());
+    assertEquals(permissions, authenticationInfo("jacksmith").getUserInfo().getPermissions());
+  }
+
+  @Test
+  public void testDirectUserPermissions() {
+    AuthenticationInfoDTO info = authenticationInfo("jacksmith");
+
+    assertNotNull(info);
+    assertTrue(info.getRbacEnabled());
+    System.out.println("info = " + info);
+    List<UserPermissionDTO> permissions = info.getUserInfo().getPermissions();
+    assertFalse(permissions.isEmpty());
+  }
+
+  @Test
+  public void testEmptyPermissions() {
+    assertTrue(Objects.requireNonNull(authenticationInfo("johnjames"))
+        .getUserInfo()
+        .getPermissions()
+        .isEmpty()
+    );
+  }
+
+  private String session(String name) {
+    return Objects.requireNonNull(
+            webTestClient
+                .post()
+                .uri("/login")
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                .body(BodyInserters.fromFormData("username", name).with("password", name + "@kafbat.io"))
+                .exchange()
+                .expectStatus()
+                .isFound()
+                .returnResult(String.class)
+                .getResponseCookies()
+                .getFirst(SESSION))
+        .getValue();
+  }
+
+  private AuthenticationInfoDTO authenticationInfo(String name) {
+    return webTestClient
+        .get()
+        .uri("/api/authorization")
+        .cookie(SESSION, session(name))
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .returnResult(AuthenticationInfoDTO.class)
+        .getResponseBody()
+        .blockFirst();
+  }
+
+  public static class Initializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+
+    @Override
+    public void initialize(ConfigurableApplicationContext context) {
+      System.setProperty("spring.ldap.urls", LDAP_CONTAINER.getLdapUrl());
+      System.setProperty("oauth2.ldap.activeDirectory", "false");
```

Comment:
perhaps redundant as mentioned in other comment

Reply:
Remove too, thanks for the catch